ostree/ostree_transport.go

```diff
@@ -0,0 +1,171 @@
+	// and "" could be unexpectedly shadowed by the "/" entry.
+	if scope == "/" {
+		return errors.New(`Invalid scope "/": Use the generic default scope ""`)
+	}
+	cleaned := filepath.Clean(scope)
+	if cleaned != scope {
+		return errors.Errorf(`Invalid scope %s: Uses non-canonical format, perhaps try %s`, scope, cleaned)
+	}
+	return nil
+}
+
+// ostreeReference is an ImageReference for ostree paths.
+type ostreeReference struct {
+	ref          string
+	repo         string
+	image        reference.Named
+}
+
+func (t ostreeTransport) ParseReference(ref string) (types.ImageReference, error) {
+	var repo = ""
+	var imageStr = ""
+	s := strings.SplitN(ref, "@", 2)
+	if len(s) == 1 {
+		imageStr, repo = s[0], "/ostree/repo"
+	} else {
+		imageStr, repo = s[0], s[1]
+	}
+
+	image, err := reference.ParseNamed(imageStr)
+	imageTagged := reference.WithDefaultTag(image)
+	if err != nil {
+		return nil, err
+	}
+	return NewReference(ref, repo, imageTagged)
+}
+
+func NewReference(reference string, repo string, image reference.Named) (types.ImageReference, error) {
+	return ostreeReference{ref: reference, repo: repo, image: image}, nil
+}
+
+func (ref ostreeReference) Transport() types.ImageTransport {
+	return Transport
+}
+
+// StringWithinTransport returns a string representation of the reference, which MUST be such that
+// reference.Transport().ParseReference(reference.StringWithinTransport()) returns an equivalent reference.
+// NOTE: The returned string is not promised to be equal to the original input to ParseReference;
+// e.g. default attribute values omitted by the user may be filled in in the return value, or vice versa.
+// WARNING: Do not use the return value in the UI to describe an image, it does not contain the Transport().Name() prefix.
+func (ref ostreeReference) StringWithinTransport() string {
+	return ref.ref
+}
+
+// DockerReference returns a Docker reference associated with this reference
+// (fully explicit, i.e. !reference.IsNameOnly, but reflecting user intent,
+// not e.g. after redirect or alias processing), or nil if unknown/not applicable.
+func (ref ostreeReference) DockerReference() reference.Named {
+	return nil
+}
+
+func (ref ostreeReference) PolicyConfigurationIdentity() string {
+	return fmt.Sprintf("%s@%s", ref.image.String(), ref.repo)
+}
+
+// PolicyConfigurationNamespaces returns a list of other policy configuration namespaces to search
+// for if explicit configuration for PolicyConfigurationIdentity() is not set.  The list will be processed
+// in order, terminating on first match, and an implicit "" is always checked at the end.
+// It is STRONGLY recommended for the first element, if any, to be a prefix of PolicyConfigurationIdentity(),
+// and each following element to be a prefix of the element preceding it.
+func (ref ostreeReference) PolicyConfigurationNamespaces() []string {
+	return policyconfiguration.DockerReferenceNamespaces(ref.image)
```

Comment:
An image is stored in a repository, the repository usually is `/ostree/repo` but when used by non root users, we have a separate ostree repository for each user in their home directory.  I've switched it to be repo@image now, that should be enough to indicate a specific image in a repo